viewing_party/party.py

```diff
@@ -0,0 +1,226 @@
+def create_movie(movie_title, genre, rating):
+    movie = {
+        "title": movie_title,
+        "genre": genre,
+        "rating": rating
+  
+    }
+
+    if movie_title is None:
+        return None
+    if genre is None:
+        return None
+    if rating is None:
+        return None
+    return movie
+
+
+def add_to_watched(user_data, movie):
+        user_data["watched"].append(movie)
+
+        return user_data
+    
+def add_to_watchlist(user_data, movie):
+        user_data["watchlist"].append(movie)
+
+        return user_data
+
+def watch_movie(user_data, title):
+        for movie in user_data["watchlist"]:
+            if movie["title"] == title:
+                user_data["watchlist"].remove(movie)
+                user_data["watched"].append(movie)
+                new_watchlist = user_data["watchlist"]
+        
+        # converted_watchlist = set(user_data["watchlist"])
+        # converted_watched = set(user_data["watched"])
+
+        # result_converted = converted_watchlist - converted_watched
+        # user_data["watchlist"] = list(result_converted)
+
```

Comment:
Be careful about modifying a collection as you're iterating through it (watchlist). For this example, we're only removing a single instance of a movie, however, if there are multiple instances of that movie that could cause the rest of the items in the list to shift position and potentially skip over other items as the loop continues.

This case is fine, but let's proceed with caution when modifying a collection while iterating through it.
Here's more info: https://thispointer.com/python-remove-elements-from-a-list-while-iterating/

Reply:
```suggestion
        for movie in user_data["watchlist"]:
            if movie["title"] == title:
                user_data["watchlist"].remove(movie)
                user_data["watched"].append(movie)

        # Start of set solution
        # converted_watchlist = set(user_data["watchlist"])
        # converted_watched = set(user_data["watched"])

        # result_converted = converted_watchlist - converted_watched
        # user_data["watchlist"] = list(result_converted)
        # End of set solution
```

In addition, I don't think you need `new_watchlist`. It's a data structure that's initialized but not used. If it's not used, I recommend removing it from your code.

Also, I see you've commented your set solution. I recommend adding a comment around the code block that labels its purpose just to prevent future confusion.